datafusion/sql/src/planner.rs

```diff
@@ -316,34 +318,40 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             Statement::Drop {
                 object_type,
                 if_exists,
-                names,
+                mut names,
                 cascade: _,
                 restrict: _,
                 purge: _,
+            } => {
                 // We don't support cascade and purge for now.
                 // nor do we support multiple object names
-            } => match object_type {
-                ObjectType::Table => Ok(LogicalPlan::DropTable(DropTable {
-                    name: names
-                        .get(0)
-                        .ok_or_else(|| ParserError("Missing table name.".to_string()))?
-                        .to_string(),
-                    if_exists,
-                    schema: DFSchemaRef::new(DFSchema::empty()),
-                })),
-                ObjectType::View => Ok(LogicalPlan::DropView(DropView {
-                    name: names
-                        .get(0)
-                        .ok_or_else(|| ParserError("Missing table name.".to_string()))?
-                        .to_string(),
-                    if_exists,
-                    schema: DFSchemaRef::new(DFSchema::empty()),
-                })),
-                _ => Err(DataFusionError::NotImplemented(
-                    "Only `DROP TABLE/VIEW  ...` statement is supported currently"
-                        .to_string(),
-                )),
-            },
+
+                let name = match names.len() {
+                    0 => Err(ParserError("Missing table name.".to_string()).into()),
+                    1 => object_name_to_table_reference(names.pop().unwrap()),
+                    _ => {
+                        Err(ParserError("Multiple objects not supported".to_string())
```

Comment:
Previously if multiple objects were specified, the second was ignored: https://github.com/apache/arrow-datafusion/issues/4531